src/streaming/node.hh

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "array.dimensions.hh"
+#include "blosc.compression.params.hh"
+#include "definitions.hh"
+#include "s3.connection.hh"
+#include "sink.hh"
+#include "thread.pool.hh"
+#include "zarr.types.h"
+
+#include <string>
+
+namespace zarr {
+struct NodeConfig
```

Comment:
The name `Node` is too generic. Some possible alternatives 'ZarrElement', 'ZarrContainer'.